Automattic/wp-calypso
# Add promote posts button to stats entries, with feature flag

#### Proposed Changes

Hi Team, this is my first calypso PR and part of a series to add entrypoints in calypso for a new post promotion widget. Which will allow wordpress.com users to pay for advertising and serve the posts as ads on various A8C products.

Info specifically on these tasks here (pbpazC-3cN-p2) and the wider project thread here (pbpazC-2Pd-p2)

The majority of this feature is done outside of WP.com as a standalone widget, here we only need to add buttons to Calypso to show the widget and to pass it the site & post id's.

Note: We will be iterating on a bunch of PRs to complete this feature, but to keep these smaller we will be merging one-by-one under a feature flag.

Here are our provided designs:

![Screenshot 2022-07-28 at 11 05 39](https://user-images.githubusercontent.com/6440498/181481571-8243cd85-8b02-4c2b-b875-2ea19810239e.png)

And here is the screenshot in my development environment:


#### Testing Instructions

-  Enable the feature flag `promote-posts` either on [startup](https://wpcalypso.wordpress.com/devdocs/config/README.md#feature-flags:~:text=Testing%20Feature%20Flags,reader%20yarn%20start) or [via url](https://wpcalypso.wordpress.com/devdocs/config/README.md#feature-flags:~:text=Testing%20Feature%20Flags%20via%20URLs,and%20disables%20feature%20bar.) using the `?flags=promote-post` query string on the following urls
-  Visit the stats page for your favourite blog for example: `http://calypso.localhost:3000/stats/day/something.wordpress.com`
-  In the posts and pages section, hover over a post and you should see a "speaker" icon like in the designs abover
![Screenshot 2022-07-28 at 11 20 45](https://user-images.githubusercontent.com/6440498/181482921-286196f3-4026-47ff-bd02-464da815454e.png)
-  clicking the button should open the widget.  However, this is not in production yet so without setting up your environment you can check the network tab and notice that it's trying to hit the script `widget.js` - this will repeatedly fail.      In a working dev environment - it only ever loads the script once.
-  Test the above Steps without the feature flag enabled,  ensure `widget.js` is not loaded in the f12 -> network tab and that the speaker button does not appear on hover.

**Without local instance of the widget running**
![Screenshot 2022-07-28 at 11 29 46](https://user-images.githubusercontent.com/6440498/181484522-9f42b8e7-9024-4011-9e6f-c04d05575213.png)

**With local instance of the widget running**
![Screenshot 2022-07-28 at 11 33 47](https://user-images.githubusercontent.com/6440498/181485261-5912b8fb-51d8-4807-94ee-a339f05990fd.png)


#### Pre-merge Checklist






Related to #
